### DISCUSSION TITLE
Is private key of Phoenix/Eclair wallet actually stored on mobile phone

### DISCUSSION THREAD

**Discussion by tonyctsiu** [ACINQ/eclair#2546] [Developers]
Dear Sirs,

Just would like to know if the private key is stored in the mobile phone by using Eclair or Phoenix wallet.

Thank you 

Tony

**Comment by t-bast** [comment]
The private key is of course stored on your device, Phoenix is non-custodial, as stated in our [FAQ}(https://phoenix.acinq.co/faq#is-phoenix-a-real-lightning-node).

There is a dedicated repository for Phoenix: https://github.com/acinq/phoenix

Here you are on the repository for eclair, which is a separate server node implementation.

**Comment by tonyctsiu** [comment]
Thanks t-bast!

**Comment by tonyctsiu** [comment]
Dear t-bast,

Just some follow up questions.
Should the private key used in Eclair/Phoenix be a different one from the private key holding the big fund of bit coin (principal account)? So that I won't lose too much even if the key is stolen.

Is Eclair/Phoenix convenient to be used with a private key different from that of my principal account?
Would I need to pay extra transaction cost if I don't use the same private key of my principal account? If yes, what would be the cost impact?

Thank you!
